src/test/java/cloud/fogbow/ras/core/OrderControllerTest.java

```diff
@@ -167,36 +146,271 @@ public void testGetInactiveOrder() throws InstanceNotFoundException {
+
+        // exercise / verify
+        this.ordersController.deleteOrder(attachmentOrder);
+        this.ordersController.deleteOrder(computeOrder);
+        this.ordersController.deleteOrder(volumeOrder);
+    }
+
+    // test case: Deletes an order with an instance, this should call
+    // cloud connector deleteOrder
+    @Test
+    public void testDeleteOrderWithInstanceRunning() throws FogbowException {
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        computeOrder.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        this.ordersController.activateOrder(computeOrder);
+
+        Mockito.doNothing().when(this.localCloudConnector).deleteInstance(Mockito.any());
+        
+        // exercise
+        this.ordersController.deleteOrder(computeOrder);
+
+        // verify
+        Mockito.verify(this.localCloudConnector, Mockito.times(1)).deleteInstance(Mockito.any());
     }
 
-    // test case: Checks if given an order getResourceInstance() returns its instance.
+    // test case: Deletes an order with an instance, and cloudConnector
+    // thrown an Exception
+    @Test(expected = FogbowException.class)
+    public void testDeleteOrderWithInstanceCloudConnectorFail() throws FogbowException {
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        computeOrder.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+        this.ordersController.activateOrder(computeOrder);
+        
+        Mockito.doThrow(FogbowException.class).when(this.localCloudConnector).deleteInstance(Mockito.any());
+
+        // exercise
+        this.ordersController.deleteOrder(computeOrder);
+    }
+
+    // test case: Checks if given an OPEN order getResourceInstance() throws
+    // RequestStillBeingDispatchedException.
+    @Test(expected = RequestStillBeingDispatchedException.class)
+    public void testGetResourceInstanceOfOpenOrder() throws FogbowException {
+        // set up
+        Order order = createRemoteOrder(getLocalMemberId());
+        order.setOrderStateInTestMode(OrderState.OPEN);
+
+        // exercise
+        this.ordersController.getResourceInstance(order);
+    }
+
+    // test case: Checks if given an order in the getResourceInstance method returns its instance.
     @Test
     public void testGetResourceInstance() throws Exception {
         // set up
-        LocalCloudConnector localCloudConnector = Mockito.mock(LocalCloudConnector.class);
+        Order order = createLocalOrder(getLocalMemberId());
+        order.setOrderStateInTestMode(OrderState.FULFILLED);
+        order.setInstanceId(BaseUnitTests.FAKE_INSTANCE_ID);
+
+        this.fulfilledOrdersList.addItem(order);
+        this.activeOrdersMap.put(order.getId(), order);
+
+        OrderInstance orderInstance = new ComputeInstance(BaseUnitTests.FAKE_INSTANCE_ID);
+        orderInstance.setState(InstanceState.READY);
+
+        Mockito.doReturn(orderInstance).when(this.localCloudConnector)
```

Comment:
The test does not work without mocking the return of getInstance method, perhaps it would be better to use verity instead of an assert equals. I'll redo the verify tests.